backend/controllers/questionController.js

```diff
@@ -36,6 +36,67 @@ exports.get_questions = [
   },
 ];
 
+// GET quiz version of questions
+export const getQuizQuestions = async (category) => {
+  let categoryId = new mongoose.Types.ObjectId(category);
+
+  const questions = await Question.aggregate([
+    {
+      $match: {
+        $and: [
+          {
+            category: categoryId,
+          },
+          {
+            options: {
+              $elemMatch: {
+                isCorrect: true,
+              },
+            },
+          },
+        ],
+      },
+    },
+    {
+      $sample: {
+        size: 10,
+      },
+    },
+    {
+      $lookup: {
+        from: 'categories',
+        localField: 'category',
+        foreignField: '_id',
+        as: 'category',
+      },
+    },
+  ]);
+
+  // Define array for the questions
+  const quizQuestions = [];
```

Comment:
should not work since const is noneditable variable. Change to let.

Reply:
const prevents reassigning the variable, but allows mutation.
As push() doesn't reassign the variable, changing it to let might cause some linters throw an error.

Reply:
Yeah that is acceptable.